tests/models/components/test_positional_embeddings.py

```diff
@@ -0,0 +1,29 @@
+import random
+
+import numpy as np
+import pytest
+
+from ami.models.components.positional_embeddings import get_2d_positional_embeddings
+
+
+class TestPositionalEmbeddings:
+    @pytest.mark.parametrize("embed_dim", [384, 768])
+    @pytest.mark.parametrize("max_grid_size", [(1), (128), (1, 1), (1, 128), (128, 1), (128, 128)])
+    def test_get_2d_positional_embeddings(
+        self,
+        embed_dim: int,
+        max_grid_size: int | tuple[int, int],
+    ):
+        grid_size: int | tuple[int, int] = (
+            random.randint(1, max_grid_size)
+            if isinstance(max_grid_size, int)
+            else (random.randint(1, max_grid_size[0]), random.randint(1, max_grid_size[1]))  # (height, width)
+        )
```

Comment:
ここは`grid_size`を直接パラメトライズするべき。ランダムにすると色々な場合をテストできていそうな気がするけど、確率的にテストが失敗して、無視していいものかどうか判別できないものになったりしてとても危ない。
そしてテストは処理内容が透明でわかりやすいことがとても重要。だから下手に複雑にしてはいけないんだ。
```suggestion
    @pytest.mark.parametrize("grid_size", [(1), (128), (1, 1), (1, 128), (128, 1), (128, 128)])
    def test_get_2d_positional_embeddings(
        self,
        embed_dim: int,
        grid_size: int | tuple[int, int],
    ):
```